mscore/toolbarEditor.cpp

```diff
@@ -106,52 +108,101 @@ void ToolbarEditor::accepted()
       Workspace::currentWorkspace->setDirty(true);
       }
 
+//---------------------------------------------------------
+//   listItem
+//---------------------------------------------------------
+QListWidgetItem* ToolbarEditor::listItem(const char* id)
+      {
+      QListWidgetItem* item = _listItem(id);
+
+      if (!item)
+            return nullptr;
+
+      // add a sensible height for cases where there is no icon
+      // to avoid row looking squashed
+      item->setSizeHint(QSize(item->sizeHint().width(), 26));
+
+      item->setData(Qt::UserRole, QVariant::fromValue((void*)id));
+      return item;
+      }
+
+QListWidgetItem* ToolbarEditor::_listItem(const char* id)
+      {
+      if (QString(id).isEmpty())
+            return new QListWidgetItem(tr("Separator"));
+
+      if (!strcmp("view-mode", id))
+            return new QListWidgetItem(tr("View mode"));
+
+      if (!strcmp("zoom-options", id))
+            return new QListWidgetItem(tr("Zoom options"));
+
+      QAction* action = getAction(id);
+
+      if (action) {
+            QString itemName = Shortcut::getShortcut(id)->text();
+
+            // in this case we can't use the menu name (it's too short
+            // as it's used for the button label). So use the shortcut
+            // description instead
+            if (strncmp(id, "voice-", 6) == 0)
+                  itemName = Shortcut::getShortcut(id)->descr();
+
+            return new QListWidgetItem(action->icon(), itemName);
+      }
+
+      qDebug()<<"ToolbarEditor does not recognize id for toolbar item: "<<QString(id);
```

Comment:
Is it OK to leave the qDebug() in or is it better to do something with ASSERT (the syntax of which I'd need to look up :)). This should never be sprung, but it's pretty benign if it is.

Reply:
IIRC gDebug()s vanish when build in RELEASE mode